### PR TITLE
feat: Add langchain support

### PR DESCRIPTION
**Pull request by amir-newrelic**

### CODE REVIEW

File: src/monitor.ts [hunk 1]
```diff
@@ -7,17 +9,17 @@ import {
...
 } from './eventDataFactory';
 
+
 export interface MonitorOpenAIOptions extends EventClientOptions {
   applicationName: string;
 }
 
 export const monitorOpenAI = (
-  openAIApi: OpenAIApi,
+  openAIApi: any,
```
Comment (nr-vsiomin) [inline src/monitor.ts:16]: We don't need this parameter anymore. The next version of the SDK should have a breaking change.

File: src/monitor.ts [hunk 2]
```diff
@@ -132,13 +134,32 @@ export const monitorOpenAI = (
...
-    openAIApi.createChatCompletion.bind(openAIApi),
-  );
-  openAIApi.createEmbedding = patchEmbedding(
-    openAIApi.createEmbedding.bind(openAIApi),
-  );
+  const createCompletion = OpenAIApi.prototype.createCompletion
+  const createChatCompletion = OpenAIApi.prototype.createChatCompletion
+  const createEmbedding = OpenAIApi.prototype.createEmbedding
+
+  const createCompletionOverRide = function (args: any) {
```
Comment (nr-vsiomin) [inline src/monitor.ts]: You have a typo, should be `createCompletionOverride`.
Same in the function bellow.

File: src/monitor.ts [hunk 3]
```diff
@@ -132,13 +134,32 @@ export const monitorOpenAI = (
...
+    return patchCompletion(
+      //@ts-ignore
```
Comment (nr-vsiomin) [inline src/monitor.ts:144]: You won't need this if you'll define it like this
`const createCompletion: any = OpenAIApi.prototype.createCompletion`

File: tests/langchain_showcase.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { OpenAI } from "langchain/llms/openai";
```
Comment (nr-vsiomin) [inline tests/langchain_showcase.ts]: Rename this file to `showcase.langchain` as its more consistent.

File: tests/monitor.unit.spec.ts [hunk 5]
```diff
@@ -15,17 +15,26 @@ const temperature = 1;
...
+    () =>
+      new Promise((resolve) => setTimeout(() => resolve(result), 1));
 
 describe('monitorOpenAI', () => {
+  //@ts-ignore
   let openai: OpenAIApi;
 
   beforeEach(() => {
+
+    //@ts-ignore
```
Comment (nr-vsiomin) [inline tests/monitor.unit.spec.ts]: The clean way to do this is to use the `__mocks__` folder.
Look at how I mocked the `@newrelic/telemetry-sdk`.

File: src/monitor.ts (hunk 1 above)
Comment (amir-newrelic) [reply src/monitor.ts:16]: we need in order it get the API key

Comment (amir-newrelic) [reply src/monitor.ts:16]: and all the configuration object

Comment (nr-vsiomin) [reply src/monitor.ts:16]: Right, so let's add proper typing `OpenAIApi | langchain`

**Review by nr-vsiomin — APPROVED** [review]

**Pull request merged**